src/terminal/osc.zig

```diff
@@ -206,12 +203,29 @@ pub const Command = union(enum) {
         report: Kind,
     };
 
-    pub const ProgressState = enum {
-        remove,
-        set,
-        @"error",
-        indeterminate,
-        pause,
+    pub const ProgressReport = struct {
+        pub const State = enum(c_int) {
+            remove,
+            set,
+            @"error",
+            indeterminate,
+            pause,
+        };
+
+        state: State,
+        progress: ?u8 = null,
+
+        pub const C = extern struct {
```

Comment:
Try to comment these of what they're synced with so its easier to find the mappings.